### PR TITLE
feat: Added category_ids and categories fields in employee_out component

### PR DESCRIPTION
**Pull request by satyamyesj**
**Description:**
Added category_ids and categories fields in employee_out component

**Click Up**
https://app.clickup.com/t/85ztmx8xr

### CODE REVIEW

File: src/components/schemas/employee.yaml [hunk 1]
```diff
@@ -203,6 +203,21 @@ employee_out:
         $ref: ./fields.yaml#/delegatees
       description: |
         List of users who are delegatees for this employee.
+    category_ids:
+      type: array
+      nullable: False
+      items:
+        $ref: './fields.yaml#/id_integer'
+      example: [ 3345, 3346 ]
+      description: |
```
Comment (modernwarfareuplink) [inline src/components/schemas/employee.yaml:212]: Screenshot 2023-08-22 at 3 23 12 PM

This is a nullable key right @satyamyesj ? 

Do we throw error when user sends category_ids but the employee category restriction is disabled ?

**Review by modernwarfareuplink — CHANGES_REQUESTED** [review]

File: src/components/schemas/employee.yaml (hunk 1 above)
Comment (satyamyesj) [reply src/components/schemas/employee.yaml:212]: @modernwarfareuplink updated API contract, this field should not be null. No we are no doing any validation depending upon category restriction is disabled or not.

Comment (modernwarfareuplink) [reply src/components/schemas/employee.yaml:212]: @satyamyesj If it is already decided then it should be fine. But all other fields in the employees invite api uses the empty array/null approach but it is different for the categories.

Screenshot 2023-08-22 at 3 52 04 PM

Comment (satyamyesj) [reply src/components/schemas/employee.yaml:212]: @modernwarfareuplink I don't undestand this, in DB we have marked project_ids, cost_center_ids as not null fields then on GET API how we are going to return null ?

Comment (modernwarfareuplink) [reply src/components/schemas/employee.yaml:212]: Seems like we don't accept `null` values for other fields and this is an error in doc.

**Review by modernwarfareuplink — APPROVED** [review]

File: src/components/schemas/employee.yaml (hunk 1 above)
Comment (satyamyesj) [reply src/components/schemas/employee.yaml:212]: @modernwarfareuplink Updated docs for other fields as well.

**Pull request merged**